### ISSUE TITLE
fatal error: stdint.h: No such file or directory

### ISSUE THREAD

**Issue by w3aryb0arpig** [sgan81/apfs-fuse#5]
I'm trying to compile this on Debian using cmake but whenever I run the the 'make' command, I receive the below compile error:-

```
username@my_system:~/apfs-fuse/build$ sudo make
/usr/bin/cmake -H/home/secuser/apfs-fuse -B/home/secuser/apfs-fuse/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/secuser/apfs-fuse/build/CMakeFiles /home/secuser/apfs-fuse/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/secuser/apfs-fuse/build'
make -f ApfsLib/CMakeFiles/ApfsLib.dir/build.make ApfsLib/CMakeFiles/ApfsLib.dir/depend
make[2]: Entering directory '/home/secuser/apfs-fuse/build'
cd /home/secuser/apfs-fuse/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/secuser/apfs-fuse /home/secuser/apfs-fuse/ApfsLib /home/secuser/apfs-fuse/build /home/secuser/apfs-fuse/build/ApfsLib /home/secuser/apfs-fuse/build/ApfsLib/CMakeFiles/ApfsLib.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/secuser/apfs-fuse/build'
make -f ApfsLib/CMakeFiles/ApfsLib.dir/build.make ApfsLib/CMakeFiles/ApfsLib.dir/build
make[2]: Entering directory '/home/secuser/apfs-fuse/build'
[  3%] Building CXX object ApfsLib/CMakeFiles/ApfsLib.dir/Aes.cpp.o
cd /home/secuser/apfs-fuse/build/ApfsLib && /usr/bin/c++    -I/home/secuser/apfs-fuse/.  -Wall -Wextra -march=native -O3 -DNDEBUG   -std=gnu++11 -o CMakeFiles/ApfsLib.dir/Aes.cpp.o -c /home/secuser/apfs-fuse/ApfsLib/Aes.cpp
In file included from /home/secuser/apfs-fuse/ApfsLib/Aes.cpp:1:0:
/usr/lib/gcc/x86_64-linux-gnu/6/include/stdint.h:9:26: fatal error: stdint.h: No such file or directory
 # include_next <stdint.h>
                          ^
compilation terminated.
ApfsLib/CMakeFiles/ApfsLib.dir/build.make:65: recipe for target 'ApfsLib/CMakeFiles/ApfsLib.dir/Aes.cpp.o' failed
make[2]: *** [ApfsLib/CMakeFiles/ApfsLib.dir/Aes.cpp.o] Error 1
make[2]: Leaving directory '/home/secuser/apfs-fuse/build'
CMakeFiles/Makefile2:88: recipe for target 'ApfsLib/CMakeFiles/ApfsLib.dir/all' failed
make[1]: *** [ApfsLib/CMakeFiles/ApfsLib.dir/all] Error 2
make[1]: Leaving directory '/home/secuser/apfs-fuse/build'
Makefile:86: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by orynider** [comment]
C:\php-sdk\phpdev\vc9\x86\php-7.2.1>nmake

Microsoft (R) Program Maintenance Utility Version 9.00.21022.08
Copyright (C) Microsoft Corporation.  All rights reserved.

Recreating build dirs
        "cl.exe" /D _USRDLL /D PHP7DLLTS_EXPORTS /D PHP_EXPORTS /D LIBZEND_EXPOR
TS /D TSRM_EXPORTS /D SAPI_EXPORTS /D WINVER=0x0601 /D COMPILE_DL_BCMATH /D COMP
ILE_DL_CALENDAR /D COMPILE_DL_COM_DOTNET /D COMPILE_DL_CTYPE /D COMPILE_DL_FILEI
NFO /D COMPILE_DL_FTP /D COMPILE_DL_HASH /D COMPILE_DL_OPCACHE /D COMPILE_DL_SHM
OP /D COMPILE_DL_SQLITE3 /D COMPILE_DL_TOKENIZER /D COMPILE_DL_ZIP /D COMPILE_DL
_EXIF /Iext/date/lib /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /DHAVE_TIMELIB_CONFIG_H
=1 /wd4244 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE
=1 -Iext/mbstring/libmbfl -Iext/mbstring/libmbfl/mbfl                   -Iext/mb
string/oniguruma /D NOT_RUBY=1 /D LIBMBFL_EXPORTS=1 /D HAVE_STDARG_PROTOTYPES=1
/D HAVE_CONFIG_H /D HAVE_STDLIB_H /D HAVE_STRICMP /D MBFL_DLL_EXPORT=1 /D EXPORT
 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /DHAVE_
SQL_H=1 /DHAVE_SQLEXT_H=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Iext/pcre/pcrelib
 -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /D HAVE_CONFIG_H /DZEND_ENABLE_STATIC_TSRML
S_CACHE=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /DHAVE_WINSOCK_H=1 /DZEND_ENABLE_S
TATIC_TSRMLS_CACHE=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /I "ext/date/lib" /DHAV
E_TIMELIB_CONFIG_H=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /DZEND_ENABLE_STATIC_TS
RMLS_CACHE=1  /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /nologo /I . /I main /I Zend /
I TSRM /I ext /D _WINDOWS /D WINDOWS=1 /D ZEND_WIN32=1 /D PHP_WIN32=1 /D WIN32 /
D _MBCS /W3 /D _USE_MATH_DEFINES /FD /wd4996 /D_USE_32BIT_TIME_T=1 /MP /LD /MD /
W3 /Ox /D NDebug /D NDEBUG /D ZEND_WIN32_FORCE_INLINE /GF /D ZEND_DEBUG=0 /D ZTS
=1 /I "C:\php-sdk\phpdev\vc9\x86\deps\include" /D FD_SETSIZE=256 /FoC:\php-sdk\p
hpdev\vc9\x86\php-7.2.1\Release_TS\main\ /FpC:\php-sdk\phpdev\vc9\x86\php-7.2.1\
Release_TS\main\ /FRC:\php-sdk\phpdev\vc9\x86\php-7.2.1\Release_TS\main\ /FdC:\p
hp-sdk\phpdev\vc9\x86\php-7.2.1\Release_TS\main\ /c main\SAPI.c main\fopen_wrapp
ers.c main\getopt.c main\internal_functions.c main\main.c main\mergesort.c main\
network.c main\output.c main\php_content_types.c main\php_ini.c main\php_open_te
mporary_file.c main\php_scandir.c main\php_sprintf.c main\php_ticks.c main\php_v
ariables.c main\reentrancy.c main\rfc1867.c main\snprintf.c main\spprintf.c main
\strlcat.c main\strlcpy.c
SAPI.c
fopen_wrappers.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
getopt.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
internal_functions.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
main.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
mergesort.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
network.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
output.c
php_content_types.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
php_ini.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
php_open_temporary_file.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
php_scandir.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
php_sprintf.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
php_ticks.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
php_variables.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
reentrancy.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
rfc1867.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
snprintf.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
spprintf.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
strlcat.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
strlcpy.c
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
.\win32/php_inttypes.h(47) : fatal error C1083: Cannot open include file: 'stdin
t.h': No such file or directory
NMAKE : fatal error U1077: 'C:\MsVS09\VC\Bin\cl.exe' : return code '0x2'
Stop.

C:\php-sdk\phpdev\vc9\x86\php-7.2.1>

**Comment by sgan81** [comment]
Well, the 'stdint.h' file is not available in MSVC 9.0, so the solution there is to write one yourself. Just write one yourself, defining uint8_t, uint16_t, uint32_t, uint64_t and it's intXX_t counterparts, and the program should be happy. Or get one off the net for MSVC 9.0.

It should be available in gcc / clang though. Maybe you are using an older gcc that needs to specify -std=c++11 on the command line? CMake should already do that though...

**Comment by sgan81** [comment]
Thinking again, the driver will not compile on msvc 9. It needs a C++ 11 compiler.
And the issue with debian, no idea, maybe missing some packets, or installation too old.

Since those are not issues with apfs-fuse itself, I'm going to close this issue.
